### PR TITLE
[wip] Adopt external tower-grpc and tower-h2 deps

### PR DESCRIPTION
**Pull request by hawkw**
_This is a work in progress and should not be merged before the Conduit 0.2 release._

This PR unvendors all the Rust dependencies except for `futures-mpsc-lossy`, which doesn't exist elsewhere, and `tower-router`, for which the vendored version is actually _ahead_ of the version in the `tower-rs/tower` repository.

As of 0fa5d51, all the proxy's tests now pass, and I can successfully deploy Conduit in Minikube with Emojivoto. Everything seems *basically* not broken, but this obviously needs some more principled testing. 

We definitely don't want to merge this to master until after the upcoming release, but I'm putting it up for review now, and hopefully we can get a head start on testing.

### CODE REVIEW

**Comment by hawkw** [comment]
@klingerf I'm adding you as a reviewer as well in the hopes that you can work your release confidence magic when you get a chance.

**Comment by hawkw** [comment]
oh: screen shot 2018-01-25 at 5 41 45 pm

File: proxy/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #![cfg_attr(feature = "cargo-clippy", allow(clone_on_ref_ptr))]
 #![cfg_attr(feature = "cargo-clippy", allow(new_without_default_derive))]
-#![deny(warnings)]
+// #![deny(warnings)]
```
Comment (carllerche) [inline proxy/src/lib.rs]: Could this be reverted? This was only intended to be temporary.

File: proxy/src/control/mod.rs [hunk 2]
```diff
@@ -217,6 +217,7 @@ where
 
 // ===== impl  EnumService =====
 
+/*
```
Comment (carllerche) [inline proxy/src/control/mod.rs]: Could all this be deleted?

**Comment by carllerche** [comment]
Work remaining to do:

* [ ] Pin all git deps to a revision.
* [ ] Delete commented out code.

File: proxy/src/control/mod.rs (hunk 2 above)
Comment (hawkw) [reply proxy/src/control/mod.rs]: Yeah, I don't think I touched that file enough to notice the big blob of commented out code. Will remove.

File: proxy/src/lib.rs (hunk 1 above)
Comment (hawkw) [reply proxy/src/lib.rs]: Yeah, totally forgot, thanks. AFAIK all the warnings on this branch have been dealt with.

**Comment by hawkw** [comment]
@carllerche totally agree we *should* be pinning our department to fixed refs, unfortunately as far as I can tell, we can't easily do that because we can't override the `tower` version that `tower-grpc`imports in the generated code

**Comment by carllerche** [comment]
@hawkw can this not be done w/ `[patch]`?

**Comment by hawkw** [comment]
@carllerche This is what we were discussing yesterday on Slack; _I_ have not been able to figure out how to do it with `[patch]` yet.

**Comment by hawkw** [comment]
As far as I can tell, my inability to override the dependencies successfully using `[patch]` is due to rust-lang/cargo#4509 (thanks Carl for pointing this out). Unclear why `[replace]` also won't work.

**Comment by hawkw** [comment]
In the process of merging this with the latest master and making everything compile-y. Not quite done but here's where I'm at so far: https://github.com/runconduit/conduit/commit/e7f33d8419f1efd80ca43973c8a4bfd020833a8b

**Comment by hawkw** [comment]
Closing in favour of #225

**Pull request closed without merging**